src/compiler/crystal/semantic/main_visitor.cr

```diff
@@ -2139,8 +2139,24 @@ module Crystal
 
           # If the loop is endless
           if endless
-            after_while_var.bind_to(while_var)
-            after_while_var.nil_if_read = while_var.nil_if_read?
+            # If a variable was re-assigned before the first break expression,
+            # the type at the end of the while body is inaccessible upon exit
```

Comment:
Thank you for the example!

But it seems the comment doesn't match the example...?

> If a variable was re-assigned before the first break expression

But here the variable was re-assigned after the first break expression, and that's the type that is inaccessible. Or am I reading this wrong?

Reply:
I'll reword the whole comment here in an hour or so.